apps/E2E/src/common/BasePage.ts

```diff
@@ -72,6 +82,48 @@ export abstract class BasePage {
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
 
+  /**
+   * For any given page object, this method automates:
+   *  - Navigating to the page by clicking its component button
+   *  - Waiting for the component page to load
+   *  - Revealing its E2E components IF a boolean flag is passed into the method
+   *
+   * This also contains error checking through its waiters, removing the extra `expect()` calls during test setup.
```

Comment:
Add:

Parameter: showE2ESection: boolean (default false) - Explanation of why we need this param